torch/csrc/jit/mobile/backport_manager.cpp

```diff
@@ -36,16 +37,23 @@ bool update_bytecode_version(
   return false;
 }
 
-void copy_non_bytecode(
+void selective_copy(
     PyTorchStreamReader& reader,
-    PyTorchStreamWriter& writer) {
+    PyTorchStreamWriter& writer,
+    const std::unordered_set<std::string>& excluded_files) {
   auto records = reader.getAllRecords();
   for (const auto& record : records) {
     // Don't copy archive `version` and archive `bytecode`
     // Archvie `version` will be written when PyTorchStreamWriter is going to
     // finalize and run writeEndOfFile()
-    if (record != kArchiveNameVersion &&
-        record.find(kArchiveNameBytecode) == std::string::npos) {
+    bool skip = false;
+    for (const auto& excluded_file : excluded_files) {
+      if (record.find(excluded_file) != std::string::npos) {
```

Comment:
is substring safe here?
otherwise, should be be strictly checking for the full path (and thus use the set's find function instead)?

Reply:
Good point. A record includes the path but expluded_file does not. Is it possible that there are two files with the same name but in different paths?

Reply:
Yeah it's a good point. Given our current model schema, it should be fine. But I agree full path will be safer. I add a bit more context here and more comment in the source code and hope it's easier to understand. `reader.getAllRecords()` will return a file list, say:
```
script_module_v4_unify/data.pkl
script_module_v4_unify/code/__torch__/___torch_mangle_5.py
script_module_v4_unify/code/__torch__/___torch_mangle_5.py.debug_pkl
script_module_v4_unify/constants/0
script_module_v4_unify/constants.pkl
script_module_v4_unify/bytecode/0
script_module_v4_unify/bytecode.pkl
script_module_v4_unify/version
```
If we want to skip writing bytecode archive, we need to skip both `script_module_v4_unify/bytecode/0` (the actual tensor data) and `script_module_v4_unify/bytecode.pkl`. We currently don't have a way to know how many tensors in total without reading `bytecode.pkl` + `bytecode` folder together. I'm looking at an easier way to load pickle file, since it's just a zip folder and it shouldn't need any pytorch related knowledge. After it's done, we can leverage it here. It can be part of the work in T90180710. Let me add it in the comment so we can keep track of it.

Reply:
Use regex to match instead in the latest commit.

Reply:
Ok internal CI doesn't like regex. Switch back with more strict substring searching